packages/slice-machine/lib/mock/Slice.ts

```diff
@@ -1,53 +1,80 @@
 
-const createEmptyMock = (
-  sliceName: string,
-  variation: VariationSM
-): Models.VariationMock => ({
-  variation: variation.id,
-  name: variation.name,
-  slice_type: snakelize(sliceName),
-  items: [],
-  primary: {},
-});
-
-export default function MockSlice(
-  sliceName: string,
-  model: SliceSM,
-  mockConfig: Record<string, Record<string, Record<string, unknown>>> // not sure about this one.
-): Models.SliceMock {
-  const variations = model.variations.map((variation) => {
-    const mock = createEmptyMock(sliceName, variation);
-    const handler: (
-      fields?: FieldsSM,
-      mocks?: Record<string, unknown>
-    ) => Models.VariationMock["primary"] = handleFields(Widgets);
-
-    const maybeMockConfig = mockConfig?.[variation.id]?.primary || {};
-    mock.primary = handler(variation.primary, maybeMockConfig);
-
-    const items: Models.VariationMock["items"] = [];
+function buildVariationMockConfig(
+  model: Variation,
+  legacyVariationMockConfig?: Partial<Record<string, Record<string, unknown>>>
+): VariationMockConfig {
+  const primaryFields =
+    model.primary &&
+    buildFieldsMockConfig(
+      model.primary,
+      legacyVariationMockConfig?.["primary"]
+    );
+  const itemFields =
+    model.items &&
+    buildFieldsMockConfig(model.items, legacyVariationMockConfig?.["items"]);
+  return {
+    nbItemsBlocks: 1,
+    primaryFields,
+    itemFields,
+  };
+}
 
-    const repeat = variation.items;
-    if (!repeat || (repeat && repeat.length === 0)) {
+export function buildSliceMockConfig(
+  model: SharedSlice,
+  legacyMockConfig: Partial<
+    Record<string, Partial<Record<string, Partial<Record<string, unknown>>>>>
+  >
+): ReadonlyArray<SharedSliceMockConfig> {
+  const variationConfigs: Record<string, VariationMockConfig> =
+    model.variations.reduce((acc, variationModel) => {
+      const variationMockConfig = legacyMockConfig[variationModel.id];
       return {
-        ...mock,
-        items,
+        ...acc,
+        [variationModel.id]: buildVariationMockConfig(
+          variationModel,
+          variationMockConfig
+        ),
       };
-    }
+    }, {});
 
-    for (let i = 0; i < Math.floor(Math.random() * 6) + 2; i++) {
-      items.push(handler(repeat, mockConfig ? mockConfig.items || {} : {}));
-    }
-
-    mock.items = items;
-    return mock;
+  return model.variations.map((v) => {
+    return {
+      variation: v.id,
+      variations: variationConfigs,
+    };
   });
+}
 
-  return variations;
+export default function MockSlice(
+  smModel: SliceSM,
+  legacyMockConfig: Record<string, Record<string, Record<string, unknown>>> // not sure about this one.
+): unknown[] {
+  const model = Slices.fromSM(smModel);
+  const sliceMockConfig = buildSliceMockConfig(model, legacyMockConfig);
+  const sliceModel: SharedSlice = {
+    ...model,
+    variations: model.variations.map((v) => {
+      return {
+        ...v,
+        imageUrl: "",
```

Comment:
what if variations already have a screenshot uploaded ?

Reply:
I don't really care here, it's about having a proper model to generate mocks, the variation imageURL is useless here.

Reply:
If we don't care and the base model can have an image urls it means that there is a problem with the types no ?
What i mean is, if the function using Variations and the type were the same, you wouldn't have to do that no ?

Reply:
yeah they could share a type but the thing is, SM and prismic don't share the exact same type so it's the same as having the fallback, one way or another we need a value here, having a URL we don't use over an empty string is not relevant here.